src/transport/raw/MessageHeader.cpp

```diff
@@ -228,7 +228,7 @@ CHIP_ERROR PacketHeader::Encode(uint8_t * data, size_t size, uint16_t * encode_s
 
     {
         // Payload flags are restricted.
-        VerifyOrExit((payloadFlags.Raw() & Header::kValidPayloadFlags) == payloadFlags.Raw(), err = CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrExit(payloadFlags.HasOnly(Header::FlagValues::kVendorIdPresent), err = CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This is losing the semantic "valid payload flags" thing...  Do we really want to do that?

Reply:
There seems to be no value to having a global constant for it. The intent seems clearer now:

payload flags is only allowed to have kVendorIdPresent in it.